flows/actions/save_contact_field.go

```diff
@@ -16,18 +16,16 @@ const TypeSaveContactField string = "save_contact_field"
 //   {
 //     "uuid": "8eebd020-1af5-431c-b943-aa670fc74da9",
 //     "type": "save_contact_field",
-//     "field_uuid": "0cb17b2a-3bfe-4a19-8c99-98ab9561045d",
-//     "field_name": "Gender",
+//     "field": {"key": "gender", "label": "Gender"},
 //     "value": "Male"
 //   }
 // ```
 //
 // @action save_contact_field
 type SaveContactField struct {
 	BaseAction
-	FieldUUID flows.FieldUUID `json:"field_uuid"    validate:"required,uuid4"`
-	FieldName string          `json:"field_name"    validate:"required"`
-	Value     string          `json:"value"`
+	Field *flows.FieldReference `json:"field" validate:"required"`
```

Comment:
Why do we use a nested object for the field in the action but use `field_key` in the event? Seems these should be consistent.

Reply:
I think pretty much all actions currently use nested objects for any sort of reference (labels, groups, field, contacts, subflows). In the label/group/field case these are things you might create on flow import and thus need extra fields. Yesterday Eric and I were discussing the possibility of including value_type with field references in flows so that import knows to create a field with a specific type.

All our events on the otherhand use single values - uuids for groups/labels/flows/contacts and in this case, key for a field. The thing being referenced in an event is assumed to exist and won't be created if it doesn't.

Action references can also sometimes include expression templates but obviously a reference to something in an event can't. So I guess I see these as different things - we should strive to be consistent across all actions and across all events, but not across both of those things.

Reply:
I don't like it but ok.

Reply:
If you think events should use the same reference objects as in the flow spec then we can discuss changing them to be like that, but for now this change is at least being consistent with how things currently are.